exercises/allergies/example.R

```diff
@@ -0,0 +1,30 @@
+allergy <- function(num) {
+  
+  allergy_list <- list(
+    cats = 128,
+    pollen = 64,
+    chocolate = 32,
+    tomatoes = 16,
+    strawberries = 8,
+    shellfish = 4,
+    peanuts = 2,
+    eggs = 1
+  )
+  
+  check_allergy <- function(allergy, val) {
+    val %% (allergy * 2) >= allergy
```

Comment:
Not a _review comment_, but a good old comprehension question: Is this line a "greedy" check whether any of the given allergy values fits into the total score? In a C tutorial over a year ago, I implemented a greedy algorithm to calculate composing a given amount of change (money) from as few as possible, as large as possible coins. The problem here kind of matches, but I'm having a hard time understanding why this line of math would solve the problem. It does, though O.o